src/models/v3/message-trait.ts

```diff
@@ -15,11 +15,11 @@ import type { v3 } from '../../spec-types';
 
 export class MessageTrait<J extends v3.MessageTraitObject = v3.MessageTraitObject> extends CoreModel<J, { id: string }> implements MessageTraitInterface {
   id(): string {
-    return this.messageId() || this._meta.id || this.extensions().get(xParserMessageName)?.value<string>() as string;
+    return this._meta.id || this.extensions().get(xParserMessageName)?.value<string>() as string;
   }
 
   hasMessageId(): boolean {
-    return !!this._json.messageId;
+    return false;
   }
```

Comment:
Why is this function not removed? 🤔

Reply:
Because we want to keep the same API regardless of the AsyncAPI version, right? We need to keep the API stable independently of the version you're using.

Reply:
Solved in https://github.com/asyncapi/parser-api/pull/102#discussion_r1382421659